File: src/main/java/com/gitlab/daring/image/command/parameter/DoubleParam.java
package com.gitlab.daring.image.command.parameter;

import static java.lang.Double.parseDouble;

public class DoubleParam extends NumberParam<Double> {

	public DoubleParam(String sv, String sp) {
		super(sv, sp);
	}

	@Override
	public Double parseValue(String sv) {
		return parseDouble(sv);
	}

	@Override
	public void setNumValue(Number v) {
		setValue(v.doubleValue());
	}
	
}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/DoubleParam.kt
package com.gitlab.daring.image.command.parameter

class DoubleParam(sv: String, sp: String) : NumberParam<Double>(sv, sp) {

    override fun parseValue(sv: String) = sv.toDouble()

    override fun setNumValue(v: Number) {
        value = v.toDouble()
    }

}